apps/cowswap-frontend/src/legacy/state/price/updater.ts

```diff
@@ -152,6 +153,9 @@ export default function FeesUpdater(): null {
   const isUnsupportedTokenGp = useIsUnsupportedTokenGp()
 
   const permitHookParams = usePermitHookParams(sellCurrency)
+  const permitHookData = usePermitHookData(permitHookParams)
```

Comment:
You did great by making a `permit` module.

But here are some ideas, to have cleaner dependencies:
- `usePermitHookData` probably should not have any parameter. It could depend on `useWallet` to update the data is returning. This way, you make everything outside of the `permit` module dumb (they will just consume the hook)
- Here, in this updater, u should have no logic regarding permit. Here w just use the `appData``, so its up to this module what the value of it should be
- `appData` module should use `usePermitHookData` to update the appData accordinly

Reply:
Thanks for the suggestions, this is what I was struggling with!

I did not consider adding the permit hook into appData as I wanted to keep it self contained.
But I guess unless I create another module that does the mix of both modules, there will need to be module imports.